lib/Driver/Driver.cpp

```diff
@@ -2167,45 +2189,58 @@ void Driver::computeMainOutput(Compilation &C, const JobAction *JA,
                                SmallVectorImpl<const Action *> &InputActions,
                                SmallVectorImpl<const Job *> &InputJobs,
                                const TypeToPathMap *OutputMap,
-                               StringRef BaseInput, llvm::SmallString<128> &Buf,
+                               StringRef BaseInput, StringRef PrimaryInput,
+                               llvm::SmallString<128> &Buf,
                                CommandOutput *Output) const {
   StringRef OutputFile;
   if (OI.isMultiThreading() && isa<CompileJobAction>(JA) &&
       types::isAfterLLVM(JA->getType())) {
     // Multi-threaded compilation: A single frontend command produces multiple
     // output file: one for each input files.
-    auto OutputFunc = [&](StringRef Input) {
+    auto OutputFunc = [&](StringRef Base, StringRef Primary) {
       const TypeToPathMap *OMForInput = nullptr;
       if (OFM)
-        OMForInput = OFM->getOutputMapForInput(Input);
+        OMForInput = OFM->getOutputMapForInput(Base);
 
       OutputFile = getOutputFilename(C, JA, OI, OMForInput, TC.getTriple(),
-                                     C.getArgs(), AtTopLevel, Input, InputJobs,
+                                     C.getArgs(), AtTopLevel, Base, Primary,
                                      Diags, Buf);
-      Output->addPrimaryOutput(OutputFile, Input);
+      Output->addPrimaryOutput(CommandInputPair{.Base=Base, .Primary=Primary},
```

Comment:
Teeeechnically this isn't valid C++11. Please just use positional arguments (here and below).

Reply:
I thought we moved to C++14! Anyway, fine, ruin all my fun 😝

Reply:
Either way, it's a C++17 feature!